documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
 ...
 ```
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4032"></a>R4032 MissingXmsErrorResponse
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : The response code {0} is defined without a x-ms-error-response.
+
+**Description** :  If defines response code 4xx or 5xx ,  x-ms-error-response:true is required. There is one exception: a HEAD operation with 404 SHOULD have x-ms-error-response:false.
```

Comment:
why 404 SHOULD have exception, could you add reason in WHY section to raise attention?

Reply:
Sure, I will confirm with the initiator of this rule